File: LeetCode-Java/Evaluate Reverse Polish Notation.java
public class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> tokenStack = new Stack<Integer>();
        for(int i = 0;i < tokens.length; i ++)
        {
            if(!isOperator(tokens[i])) 
                tokenStack.push(Integer.valueOf(tokens[i]));
            else
            {
                String operator = tokens[i];
                int right = tokenStack.pop();
                int left = tokenStack.pop();
                int result = compute(operator,left,right);
                tokenStack.push(result);
            }
        }
        return tokenStack.pop();
    }
    
    boolean isOperator(String c)
    {
        if( c.compareTo("+") == 0 
            || c.compareTo("-") == 0 
            || c.compareTo("*") == 0 
            || c.compareTo("/") == 0)
            return true;
        else
            return false;
    }
    
    int compute(String c, int left, int right)
    {
        int result = 0;
        if(c.compareTo("+") == 0 )
        {
            result = left + right;
        }
        else if(c.compareTo("-") == 0 )
        {
            result = left - right;
        }
        else if(c.compareTo("*") == 0 )
        {
            result = left * right;
        }
        else
        {
            result = left / right;
        }
        return result;
    }
}

File: Leetcode-cpp/evaluate_reverse_polish_notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> record;
        for(auto token : tokens) {
            if (token.compare("+") == 0) {
                int v1 = record.top();
                record.pop();
                int v2 = record.top();
                record.pop();
                record.push(v1 + v2);
            } else if(token.compare("-") == 0) {
                int v1 = record.top();
                record.pop();
                int v2 = record.top();
                record.pop();
                record.push(v2 - v1);
            } else if(token.compare("*") == 0) {
                int v1 = record.top();
                record.pop();
                int v2 = record.top();
                record.pop();
                record.push(v1 * v2);
            } else if(token.compare("/") == 0) {
                int v1 = record.top();
                record.pop();
                int v2 = record.top();
                record.pop();
                record.push(v2 / v1);
            } else {
                record.push(stoi(token));
            }
        }
        return record.top();
    }
};